src/legacy/core_plugins/vega/public/vega_request_handler.js

```diff
@@ -17,28 +17,33 @@
  * under the License.
  */
 
+import chrome from 'ui/chrome';
+import { timefilter } from 'ui/timefilter';
+import { buildEsQuery, getEsQueryConfig } from '@kbn/es-query';
+
 import { VegaParser } from './data_model/vega_parser';
 import { SearchCache } from './data_model/search_cache';
 import { TimeCache } from './data_model/time_cache';
-import { timefilter } from 'ui/timefilter';
-import { buildEsQuery, getEsQueryConfig } from '@kbn/es-query';
 
-export function VegaRequestHandlerProvider(es, serviceSettings, config, esShardTimeout) {
-  const searchCache = new SearchCache(es, { max: 10, maxAge: 4 * 1000 });
-  const timeCache = new TimeCache(timefilter, 3 * 1000);
+export async function createVegaRequestHandler() {
 
+  // todo: work in progress.
+  const $injector = await chrome.dangerouslyGetActiveInjector();
+  const es = $injector.get('es');
```

Comment:
@Luke @streamich
I need somehow to get the client of 'elasticsearch' without using of AngularJS injector. Please help me with that.